### PR TITLE
Give Structured Tools Agent access to memory

### PR DESCRIPTION
**Pull request by d3n7**
Seems to make sense since it works with chat models.

### CODE REVIEW

File: langchain/agents/structured_chat/base.py [hunk 1]
```diff
@@ -82,12 +83,13 @@ def create_prompt(
         tool_names = ", ".join([tool.name for tool in tools])
         format_instructions = format_instructions.format(tool_names=tool_names)
         template = "\n\n".join([prefix, formatted_tools, format_instructions, suffix])
+        if input_variables is None:
+            input_variables = ["input", "chat_history", "agent_scratchpad"]
         messages = [
             SystemMessagePromptTemplate.from_template(template),
+            MessagesPlaceholder(variable_name="chat_history"),
```
Comment (skcoirz) [inline langchain/agents/structured_chat/base.py:90]: why do we need this? The chat history is recorded in system and human with memory. I’m confused here.

Comment (d3n7) [reply langchain/agents/structured_chat/base.py:90]: Nope, if you try to init this type of agent with a memory you will get an error, cause it’s currently zero shot

**Review by hwchase17 — COMMENTED** [review]
this should be optional - ill change

**Comment by medram** [comment]
is there a way to use `structured-chat-zero-shot-react-description` agent with memory?

**Pull request merged**